tensorbay/opendataset/SegTrack/loader.py

```diff
@@ -0,0 +1,120 @@
+    """Dataloader of the `SegTrack`_ dataset.
+
+    .. _SegTrack: http://cpl.cc.gatech.edu/projects/SegTrack/
+
+    The file structure of SegTrack looks like::
+
+        <path>
+            birdfall2/
+                birdfall2_00018.png
+                ...
+                ground-truth/
+                    birdfall2_00018.png
+                    ...
+            cheetah/
+                chasedeer_frame_0001.bmp
+                ...
+                ground-truth/
+                    chasedeer_00000.png
+                    ...
+            girl/
+                5117-8_70161.bmp
+                ...
+                ground-truth/
+                    0.bmp
+                    ...
+            monkeydog/
+                195.bmp
+                ...
+                ground-truth/
+                    Comp_00195.png
+                    ...
+            parachute/
+                parachute_00000.png
+                ...
+                ground-truth/
+                    parachute_00000.png
+                    ...
+            penguin/
+                penguin_00000.bmp
+                ...
+                ground_truth/
+                    penguin_00000.png
+                    ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.join(os.path.abspath(os.path.expanduser(path)))
+    dataset = Dataset(DATASET_NAME)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_name in _SEGMENT_NAMES:
+        segment = dataset.create_segment(segment_name)
+        segment_directory = os.path.join(root_path, segment_name)
+        mask_directory = os.path.join(segment_directory, "masks")
+        os.makedirs(mask_directory, exist_ok=True)
+        mask_name = "ground_truth" if segment_name == "penguin" else "ground-truth"
+
+        for image_path in glob(os.path.join(segment_directory, "*.*")):
+            data = Data(image_path)
+            filename = os.path.splitext(os.path.basename(image_path))[0]
+            mask_path = os.path.join(mask_directory, f"{filename}.png")
+            if not os.path.exists(mask_path):
+                mask = np.array(
+                    Image.open(
+                        _get_original_mask_path(
+                            os.path.join(segment_directory, mask_name), segment_name, filename
+                        )
+                    ),
+                    dtype=int,
+                )[:, :, 0]
+                # reformat mask
+                mask = (mask - 1) // 254 + 1
+                Image.fromarray(mask, mode="L").save(mask_path)
+            data.label.semantic_mask = SemanticMask(mask_path)
+            segment.append(data)
+    return dataset
+
+
+def _get_original_mask_path(original_mask_directory: str, segment_name: str, filename: str) -> str:
+    if segment_name == "monkeydog":
```

Comment:
you need to check the segment name every time you enter this function which is redundant